api/wasm/cpp/proxy_wasm_impl.h

```diff
@@ -997,15 +997,18 @@ struct MetricTag {
 
 struct MetricBase {
   MetricBase(MetricType t, const std::string& n) : type(t), name(n) {}
-  MetricBase(MetricType t, const std::string& n, std::vector<MetricTag> ts)
-      : type(t), name(n), tags(ts.begin(), ts.end()) {}
+  MetricBase(MetricType t, const std::string& n, std::vector<MetricTag> ts, std::string  fs=".", std::string vs=".")
```

Comment:
Could you remove the default arguments here and add back the version without the additional arguments (i.e. have 2 constructors one without fs and vs and one with).

That way the "." and "," constants appear only in one place (as the default initializers next to the definition of the instance variables).